File: cakk-api/src/main/java/com/cakk/api/controller/shop/AdminController.java
package com.cakk.api.controller.shop;

import jakarta.validation.Valid;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PatchMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import lombok.RequiredArgsConstructor;

import com.cakk.api.dto.request.shop.CreateShopRequest;
import com.cakk.api.dto.request.shop.PromotionRequest;
import com.cakk.api.dto.response.shop.CakeShopCreateResponse;
import com.cakk.api.dto.response.shop.CakeShopOwnerCandidateResponse;
import com.cakk.api.service.shop.ShopService;
import com.cakk.common.response.ApiResponse;

@RestController
@RequiredArgsConstructor
@RequestMapping("/admin")
public class AdminController {

	private final ShopService shopService;

	@GetMapping("/shops/candidates")
	public ApiResponse<CakeShopOwnerCandidateResponse> getBusinessOwnerCandidates() {
		final CakeShopOwnerCandidateResponse response =  shopService.getBusinessOwnerCandidates();

		return ApiResponse.success(response);
	}

	@PostMapping("/shops/create")
	public ApiResponse<CakeShopCreateResponse> createByAdmin(
		@Valid @RequestBody CreateShopRequest createShopRequest
	) {
		final CakeShopCreateResponse response = shopService.createCakeShopByCertification(createShopRequest);

		return ApiResponse.success(response);
	}

	@PutMapping("/shops/promote")
	public ApiResponse<Void> promoteUser(
		@Valid @RequestBody PromotionRequest promotionRequest
	) {
		shopService.promoteUserToBusinessOwner(promotionRequest);
		return ApiResponse.success();
	}
}

File: cakk-api/src/main/kotlin/com/cakk/api/controller/shop/AdminController.kt
package com.cakk.api.controller.shop

import jakarta.validation.Valid

import org.springframework.web.bind.annotation.*

import com.cakk.api.dto.request.shop.CreateShopRequest
import com.cakk.api.dto.request.shop.PromotionRequest
import com.cakk.core.dto.response.shop.CakeShopCreateResponse
import com.cakk.core.dto.response.shop.CakeShopOwnerCandidateResponse
import com.cakk.core.dto.response.shop.CakeShopOwnerCandidatesResponse
import com.cakk.api.mapper.supplyCreateShopParamBy
import com.cakk.api.mapper.supplyPromotionParamBy
import com.cakk.core.service.shop.ShopService
import com.cakk.common.response.ApiResponse

@RestController
@RequestMapping("/admin")
class AdminController(
	private val shopService: ShopService
) {

	@GetMapping("/shops/candidates")
	fun businessOwnerCandidates(): ApiResponse<CakeShopOwnerCandidatesResponse> {
		val response = shopService.getBusinessOwnerCandidates()

		return ApiResponse.success(response)
	}

    @GetMapping("/shops/candidates/{userId}")
    fun getCandidateSpecificationInformation(
		@PathVariable userId: Long
	): ApiResponse<CakeShopOwnerCandidateResponse> {
        val response = shopService.getCandidateInformation(userId)

        return ApiResponse.success(response)
    }

    @PostMapping("/shops/create")
    fun createByAdmin(
		@RequestBody @Valid request: CreateShopRequest
    ): ApiResponse<CakeShopCreateResponse> {
		val param = supplyCreateShopParamBy(request);
        val response = shopService.createCakeShopByCertification(param)

        return ApiResponse.success(response)
    }

    @PutMapping("/shops/promote")
    fun promoteUser(
		@RequestBody @Valid request: PromotionRequest
    ): ApiResponse<Unit> {
		val param = supplyPromotionParamBy(request)
        shopService.promoteUserToBusinessOwner(param)

		return ApiResponse.success()
    }
}

